packages/frontend/core/src/components/page-list/docs/page-list-header.tsx

```diff
@@ -175,24 +185,26 @@ export const TagPageListHeader = ({
   );
 };
 
-const filterOption = (option: Tag, inputValue?: string) => {
+const filterOption = (tagMeta: TagMeta, inputValue?: string) => {
   const trimmedValue = inputValue?.trim().toLowerCase() ?? '';
-  const trimmedOptionValue = option.value.trim().toLowerCase();
+  const trimmedOptionValue = tagMeta.title.trim().toLowerCase();
   return trimmedOptionValue.includes(trimmedValue);
 };
 
 interface TagsEditorProps {
-  options: Tag[];
+  tagMetas: TagMeta[];
   onClick: (open: boolean) => void;
 }
 
-export const TagsEditor = ({ options, onClick }: TagsEditorProps) => {
+export const TagsEditor = ({ tagMetas, onClick }: TagsEditorProps) => {
   const t = useAFFiNEI18N();
   const [inputValue, setInputValue] = useState('');
   const filteredOptions = useMemo(
     () =>
-      options.filter(o => (inputValue ? filterOption(o, inputValue) : true)),
-    [inputValue, options]
+      tagMetas.filter(tagMeta =>
```

Comment:
you can put the filter function into TagService

```ts
class TagService {
  filterByName(name: string) {
    this.tags.map((tags) => {
      ...do filter
      return tags
    })
  }
}
```

then this `<TagsEditor/>` dont need `tagMetas` props